agent/agent.go

```diff
@@ -4264,6 +4264,8 @@ func (a *Agent) reloadConfigInternal(newCfg *config.RuntimeConfig) error {
 
 	a.proxyConfig.SetUpdateRateLimit(newCfg.XDSUpdateRateLimit)
 
+	a.config.EnableDebug = newCfg.EnableDebug
```

Comment:
I think this would create a race condition, as we are writing `a.config.EnableDebug` in `reloadConfig` while we are reading it in the http handler. We will need to make this an atomic.

Reply:
this is inside a lock i guess, in line 4165 we have - `a.stateLock.Lock()`. Is it not atomic?

Reply:
@dhiaayachi ping